### PR TITLE
Update Company Page Cypress test Stub to use intercept

### PR DESCRIPTION
**Pull request by mendelB**
### What's this PR do?

This pull request updates the stubbed POST requests in the company-page Cypress tests which were using the [deprecated](https://docs.cypress.io/api/commands/route.html) `cy.route()` and thus actually hitting our API (which was resulting in errors per an open issue discussed in [Slack](https://dosomething.slack.com/archives/CUQMTP0RM/p1615483885001600)) to use [`cy.intercept()`](https://docs.cypress.io/api/commands/intercept.html).

### How should this be reviewed?
👀 

### Any background context you want to provide?
TL;DR: due to an unrelated update in Northstar, these scholarship POST requests started failing due to a CORS issue. This caused this test to fail (which seemed strange https://github.com/DoSomething/phoenix-next/pull/2570#issuecomment-796910304) but turns out it's bc it's actually hitting our API!

I'll add a Ghost Inspector test to cover this submission so that we have live test coverage. This one should be stubbed since it's solely testing the Phoenix functionality.

### CODE REVIEW

**Review by lindsayrmaher — APPROVED** [review]
love cypress test improvements!!
